tracing/src/macros.rs

```diff
@@ -890,7 +944,7 @@ macro_rules! enabled {
                     ":",
                     line!()
                 ),
-                kind: $crate::metadata::Kind::HINT,
+                kind: $kind,
```

Comment:
it occurs to me that we should really move the setting of the `HINT` bit to _here_, rather than having `event_enabled!` and `span_enabled!` call `.hint()` on the callsite kind. This would ensure that _all_ callsites constructed by `enabled!` invocations are _always_ hints. In the current approach, where the user can pass an arbitrary `Kind`, it's possible to call `enabled!` with a callsite that claims to be a span or event and _not_ a hint, which is incorrect. 

if we change this to
```suggestion
                kind: $kind.hint(),
```
it will ensure that the metadata always has the `HINT` callsite kind, in addition to being a span or event.

Reply:
oh i guess you are right, calling `.hint()` on something that is already a HINT is safe!